### PR TITLE
Tests & Rewarder fixes and updates

### PR DESCRIPTION
**Pull request by syahirAmali**

### CODE REVIEW

File: contracts/EscrowNative.sol [hunk 1]
```diff
@@ -53,7 +53,7 @@ contract EscrowNative is IEscrow {
 
         require(msg.value == paymentAmount + referralTax + platformTax, "Invalid amount sent");
 
-        IRewarder(rewarder).handleSeekerTaxNative{ value: referralTax + platformTax }(_seekerId, referralTax, platformTax);
+        IRewarder(rewarder).handleSeekerTaxNative{ value: referralTax + platformTax }(_seekerId, _solverId, referralTax, platformTax);
```
Comment (Cosmodude) [inline contracts/EscrowNative.sol:56]: why do we need solver id when taxing seeker?

File: contracts/EscrowNative.sol [hunk 2]
```diff
@@ -72,7 +72,7 @@ contract EscrowNative is IEscrow {
     /**
      * @notice process the dispute resolution
      */
-    function processResolution(uint8 solverShare) external onlyQuest {
+    function processResolution(uint32 solverShare) external onlyQuest {
```
Comment (Cosmodude) [inline contracts/EscrowNative.sol:75]: > using basis points now 

basis points?

File: contracts/Rewarder.sol [hunk 3]
```diff
@@ -36,31 +36,31 @@ contract Rewarder is IRewarder, Pausable {
         address token
     );
 
-    event seekerTaxPayedNative(
+    event SeekerTaxPaidNative(
         address indexed seekerAccount,
```
Comment (Cosmodude) [inline contracts/Rewarder.sol:40]: thanks

File: contracts/Rewarder.sol [hunk 4]
```diff
@@ -142,11 +142,11 @@ contract Rewarder is IRewarder, Pausable {
             rewardReferrers(solverHandler, referralTax, taxRateDivisor, address(0));
         
             // Platform tax distribution
-            address platformTaxReceiver = taxManager.platformTaxReceiver();
+            address platformTaxReceiver = taxManager.platformRevenuePool();
```
Comment (Cosmodude) [inline contracts/Rewarder.sol:145]: updated names, nice

File: contracts/Rewarder.sol [hunk 5]
```diff
@@ -244,23 +254,27 @@ contract Rewarder is IRewarder, Pausable {
...
 
         // Platform tax distribution
-        address platformTaxReceiver = taxManager.platformTaxReceiver();
+        address platformTaxReceiver = taxManager.platformRevenuePool();
         _processPayment(platformTaxReceiver, address(0), _platformTax);
 
         // Referral tax distribution
         // Rewards referrers based on referral tax value from derived payment amount from Escrow
-        rewardReferrers(seekerHandler, _referralTax, taxRateDivisor, address(0));
+        rewardReferrers(solverHandler, _referralTax, taxRateDivisor, address(0));
```
Comment (Cosmodude) [inline contracts/Rewarder.sol:272]: why do we reward solver referrals here?

File: contracts/Rewarder.sol [hunk 6]
```diff
@@ -278,43 +292,57 @@ contract Rewarder is IRewarder, Pausable {
...
         address seekerHandler = nexus.getHandler(_seekerId);
-        emit seekerTaxPayedToken(seekerHandler, msg.sender,  _platformTax + _referralTax, token);
+        address solverHandler = nexus.getHandler(_solverId);
+
+        emit SeekerTaxPaidToken(seekerHandler, msg.sender,  _platformTax + _referralTax, token);
         // Seeker tax distribution
 
         // Platform tax distribution
-        address platformTaxReceiver = taxManager.platformTaxReceiver();
+        address platformTaxReceiver = taxManager.platformRevenuePool();
```
Comment (Cosmodude) [inline contracts/Rewarder.sol:304]: we reward seeker referrals (they brought person that spends money to the platform)

File: contracts/EscrowNative.sol (hunk 2 above)
Comment (Cosmodude) [reply contracts/EscrowNative.sol:75]: oh, got it using base devisor fro deposit, good practice

File: contracts/Rewarder.sol [hunk 7]
```diff
@@ -441,78 +455,78 @@ contract Rewarder is IRewarder, Pausable {
         address token
     ) internal {
         ITaxManager taxManager = getTaxManager();
-        address[5] memory referrals; // Used to store above referrals, saving variable space
-        uint256[5] memory rewards;
+        address[4] memory referrals; // Used to store above referrals, saving variable space
```
Comment (Cosmodude) [inline contracts/Rewarder.sol:458]: right, we have 4 layers

File: contracts/TaxManager.sol [hunk 8]
```diff
@@ -15,33 +15,22 @@ contract TaxManager is ITaxManager {
...
-    uint256 public protocolTaxRate;
 
-    // Treasury addresses
-    address public platformTreasury;
-    address public disputeFeesTreasury;
+    // Pools
+    address public platformTreasuryPool;
+    address public platformRevenuePool;
+    address public referralTaxReceiver;
+    address public disputeFeesTreasuryPool;
```
Comment (Cosmodude) [inline contracts/TaxManager.sol:29]: >disputeFeesTreasuryPool
let's keep it as 
disputeFeesTreasury

File: contracts/TaxManager.sol [hunk 9]
```diff
@@ -15,33 +15,22 @@ contract TaxManager is ITaxManager {
...
 
     // Tax Rates variables
     uint256 public disputeDepositRate;  // with base divisor
-    uint256 public protocolTaxRate;
 
-    // Treasury addresses
-    address public platformTreasury;
-    address public disputeFeesTreasury;
+    // Pools
+    address public platformTreasuryPool;
```
Comment (Cosmodude) [inline contracts/TaxManager.sol:26]: and platformTreasury here

File: contracts/TaxManager.sol [hunk 10]
```diff
@@ -15,33 +15,22 @@ contract TaxManager is ITaxManager {
...
+    address public platformRevenuePool;
+    address public referralTaxReceiver;
```
Comment (Cosmodude) [inline contracts/TaxManager.sol:28]: referralTaxTreasury

**Review by Cosmodude — CHANGES_REQUESTED** [review]

File: contracts/EscrowNative.sol (hunk 1 above)
Comment (syahirAmali) [reply contracts/EscrowNative.sol]: Changes reverted

**Comment by syahirAmali** [comment]
Changes
- Reverted handleSeekerTax to use reward seeker referrals and not solver
- TaxManager pool addresses rename
- Rewarder `rewardReferrers` fix, no longer tries to send tokens to address(0) when there are no referrers

**Comment by syahirAmali** [comment]
Updates
- Reentrancy guards for native functions
- Rewarder tax unit test updates

**Review by Cosmodude — APPROVED** [review]
All good, changes on Rewarder accepted: new private function, to handleReward 

+ reentrancy

**Pull request merged**